presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -58,23 +65,37 @@ public WindowOperatorFactory(
                 List<Integer> outputChannels,
                 List<WindowFunctionDefinition> windowFunctionDefinitions,
                 List<Integer> partitionChannels,
+                List<Integer> preGroupedChannels,
                 List<Integer> sortChannels,
                 List<SortOrder> sortOrder,
+                int preSortedChannelPrefix,
```

Comment:
I find it a bit strange to be passing in the count of the sort channels that is presorted.  I think it might be easier to understand if you just passed in the presorted channels and have this code calculate the prefix size.  I don't feel strongly about this so feel free to leave it.